### PR TITLE
bug: update to fix windows failing ci validation

### LINKED ISSUE #4058: remove `add-path` from GitHub Actions
`add-path` is being deprecated and should be updated based on recommendations as shown on this week's weekly run https://github.com/microsoft/fast/actions/runs/319308716

Screen Shot 2020-10-21 at 8 24 10 AM

### PR DESCRIPTION
**Pull request by Ibrahimmaga**
# Description
  Cross Platform Validation

    Changed made to
   1. modified prettier:diff to prettier check and write in the package.json file
       added run prepare workspace.
   2. ON windows we have added this script to resolve the line ending
        run: |
          git config --global core.autocrlf false
          git config --global core.eol lf
          we have also specify the shell script as bash cmd in the workflow so the string can be treated as in Linux
          shel: bash
   3. dangerouslySetInnerHTML={{ __html: FooterTemplate }}  get replaced by 
       dangerouslySetInnerHTML={{ __html: FooterTemplate() }} to resolve a units test that was failing.
       changed made in sites/fast-color-explorer/app/site-footer.tsx
## Motivation & context

This fix os validation on windows and mac
closes #4058 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Review by EisenbergEffect — APPROVED** [review]
Approved pending review by @awentzel

File: beachball.config.js
Comment (nicholasrice) [inline beachball.config.js:1]: should this be deleted?

File: .prettierrc
Comment (nicholasrice) [inline .prettierrc:12]: nice

File: .github/workflows/ci-weekly.yml
Comment (chrisdholt) [inline .github/workflows/ci-weekly.yml:6]: Does this branch need to persist beyond this initial work?

File: sites/fast-color-explorer/app/site-footer.tsx
Comment (chrisdholt) [inline sites/fast-color-explorer/app/site-footer.tsx:16]: Is this related to the windows validation issues?

File: sites/fast-website/rollup.config.js
Comment (chrisdholt) [inline sites/fast-website/rollup.config.js:4]: Is this related to the windows validation issues?

File: beachball.config.js
Comment (Ibrahimmaga) [reply beachball.config.js:1]: let not deleted. I will work beachball deployment later on

File: sites/fast-website/rollup.config.js
Comment (Ibrahimmaga) [reply sites/fast-website/rollup.config.js:4]: yes. When we imported jst, it did fixed some issues on windows validation

File: sites/fast-color-explorer/app/site-footer.tsx
Comment (Ibrahimmaga) [reply sites/fast-color-explorer/app/site-footer.tsx:16]: This also for windows validation. John have more understanding of this than me.

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml:6]: persistence is not really useful

File: sites/fast-color-explorer/app/site-footer.tsx
Comment (radium-v) [reply sites/fast-color-explorer/app/site-footer.tsx:16]: @chrisdholt The package `rollup-plugin-ejs` has `node-sass` as a dependency, and that fails to install properly on Windows environments. The `rollup-plugin-jst` package provides the functionality we need without that dependency, but the calls here need to be referenced as functions instead of constants.

**Review by chrisdholt — APPROVED** [review]

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml]: I just remove windows-fails-ci-validation branch so we can keep the master clean

**Review by radium-v — APPROVED** [review]

File: .github/workflows/ci-weekly.yml
Comment (awentzel) [inline .github/workflows/ci-weekly.yml:1]: Let's keep the name.  It's important to distinguish this from the others.

Comment (awentzel) [reply .github/workflows/ci-weekly.yml:1]: Please update this PR with description and it looks like the check boxes in that detail are misformatted as well.

File: .github/workflows/ci-weekly.yml
Comment (awentzel) [inline .github/workflows/ci-weekly.yml:4]: I want to continue and run this only on cron-job once per week.  Please re-add schedule

File: .github/workflows/ci-weekly.yml
Comment (awentzel) [inline .github/workflows/ci-weekly.yml:17]: For readability, please add new lines after all `runs-on: ...` and after each task within each step.  There are a few that are not consistent when viewing the final version that will be merged.

File: .github/workflows/ci-weekly.yml
Comment (awentzel) [inline .github/workflows/ci-weekly.yml:22]: add a line break between `uses:..` and `- name:...`

File: .github/workflows/ci-weekly.yml
Comment (awentzel) [inline .github/workflows/ci-weekly.yml:79]: add a line break between tasks.

**Review by awentzel — CHANGES_REQUESTED** [review]
@Ibrahimmaga this looks great.  I just have a few comments on things we'll want to re-incorporate and a few nits for better readability.

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml:1]: I just change the tittle

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml:4]: ok. I will change the to on schedule

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml:17]: okay

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml:22]: done

File: .github/workflows/ci-weekly.yml
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml:79]: done

**Review by awentzel — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

**Pull request merged**
